### PR TITLE
New list models

### LINKED ISSUE #1288: [component] Add more data models for List component
- [x] Call list from fritzbox (use PHP backend from plugin)
- [x] Data from Rss/-log plugin (use PHP backend from plugin)
- [x] document new features

### PR DESCRIPTION
**Pull request by peuter**
closes #1288

### CODE REVIEW

**Comment by ChristianMayer** [comment]
Does this PR need an update due to the switch to Qx7?

**Comment by peuter** [comment]
I did a rebase here now and found some lint errors regarding the last changes in the qx7 branch. so I enabled that the ci lint fails when there are linter errors (had been disabled to weit for qx7, now we can use it again).

File: source/resource/visu_config_tile.xsd [hunk 1]
```diff
@@ -767,6 +767,24 @@
               </xsd:simpleType>
             </xsd:attribute>
             <xsd:attribute name="filter" type="xsd:string"/>
+            <xsd:attribute name="class">
+              <xsd:annotation>
+                <xsd:documentation xml:lang="en">Special builtin list models that are included in the CometVisu.</xsd:documentation>
+                <xsd:documentation xml:lang="de">Spezielles Listenmodell, die die CometVisu mitliefert.</xsd:documentation>
```
Comment (ChristianMayer) [inline source/resource/visu_config_tile.xsd:773]: Grammatik-Fehler

File: source/resource/visu_config_tile.xsd [hunk 2]
```diff
@@ -767,6 +767,24 @@
...
+                <xsd:restriction base="xsd:string">
+                  <xsd:enumeration value="FritzCallList"/>
+                  <xsd:enumeration value="RssLog"/>
+                </xsd:restriction>
+              </xsd:simpleType>
+            </xsd:attribute>
+            <xsd:attribute name="parameters" type="xsd:string">
+              <xsd:annotation>
+                <xsd:documentation xml:lang="en">Parameter for the class list models.</xsd:documentation>
+                <xsd:documentation xml:lang="de">Parameter for die class Listenmodelle.</xsd:documentation>
```
Comment (ChristianMayer) [inline source/resource/visu_config_tile.xsd:785]: `for`

File: doc/manual/de/config/structure-tile/components/list.rst [hunk 3]
```diff
@@ -261,6 +261,66 @@ Das Modell für die Mitglieder eines GroupItems, hat folgende Struktur:
         "active": true
     }]
 
+**Mitgelieferte Modelle**
+
+Die CometVisu liefert einige vordefinierte Modell mit, mit denen einige Datenquellen von Plugins eingebunden werden
```
Comment (ChristianMayer) [inline doc/manual/de/config/structure-tile/components/list.rst:266]: Modell -> Modelle

Comment (ChristianMayer) [reply doc/manual/de/config/structure-tile/components/list.rst:266]: das zweite "einige" besser entfernen

File: doc/manual/de/config/structure-tile/components/list.rst [hunk 4]
```diff
@@ -261,6 +261,66 @@ Das Modell für die Mitglieder eines GroupItems, hat folgende Struktur:
...
+können. Bisher sind das die Anrufliste einer Fritz!Box über das :doc:`tr064 <../../widgets/plugins/tr064/index>`-Plugin und die Daten aus der
```
Comment (ChristianMayer) [inline doc/manual/de/config/structure-tile/components/list.rst:267]: "Bisher sind das die Anrufliste" klingt komisch

File: doc/manual/de/config/structure-tile/components/list.rst [hunk 5]
```diff
@@ -261,6 +261,66 @@ Das Modell für die Mitglieder eines GroupItems, hat folgende Struktur:
...
+:doc:`RSSlog <../../widgets/plugins/rsslog/index>` -Datenbank. Hierfür ist es nicht nötig die Plugins in der Config einzubinden, denn
+es werden nur die Daten abgefragt und ansonsten nichts von den Plugins selbst benutzt.
```
Comment (ChristianMayer) [inline doc/manual/de/config/structure-tile/components/list.rst:269]: Was ist mit der Hidden-Config? Die muss doch vermutlich korrekt befüllt sein, denn sonst weiß die Plugin-Datenquelle ja nicht wo sie nachfragen soll

**Review by ChristianMayer — CHANGES_REQUESTED** [review]
Apart from the comments we should discuss a general design decision: what is a plugin and what isn't.

The idea behind a plugin is (was) to have "fat" code that not everyone needs separated, so that only those who need it have to pay the price for it (load time, data and memory consumption). This also includes cases where you need more than the minimum setup - which traditionally was a static web server that has no PHP. (The "ideal" low end platform was a patched Fritz!Box).

This is the reason why the TR064/FritzBox and RSSLog are plugins - they can not live without PHP and also not everyone needs the functionality.
Including the "listmodel"s for those in the main code and not in a plugin is eroding the rule set for plugins. 
When those should stay in the main code then the consequence should be that TR064 and RSSLog are also moved to the main code and their status as a plugin should be stopped.

**Comment by peuter** [comment]
I am aware of the reasons for moving code into a plugin. The tile-structure currently does not support any of the existing plugins and I currently don't have any plans to change that (although that decision might change in the future). If it is useful I would prefer to provide that functionality "built-in". There are other ways to keep avoid bloating up the code base. For example the Tile-Charts only load the library for rendering a chart if you use it, so there is no need for the user to explicitly "activate" a plugin like its done in the pure-structure. And that is what I want to achieve here -> avoiding that extra step for the user. Also I think charts are a basic requirement for modern smart-home UIs so its a builtin feature of the tile structure. Loading data from external sources is another basic requirement (and that is what the list-models are made for). They do not add much code, so there is no need to move them to an extra peace of code that need to be additionally loaded.

So the only remaining valid reason to move the listmodels in an plugin, is that you need PHP support for that. If you use openHABs internal webserver for example these list models will not work because of that. But that can be detected automatically. I am preparing another pull request with changes for openhab specific stuff (also one for openhabs codebase). That one will enable detecting that the CometVisu is delivered by openHAB. When that is merged, the next step would be to inform the user about that when he tries to use the list models (and of course also document that information) and in addition detecting that there is no PHP support (in non openhab use case) by checking the response of the already existing request to environment.php. Although the last step could already be done right now, I would prefer to do it in the order I just described here.

So long story short: The tile-structure does not support plugins (on purpose), so I cannot move it to one. And the check for PHP-Support that is needed for these list-models will be delivered in future pull requests.

**Review by ChristianMayer — APPROVED** [review]
